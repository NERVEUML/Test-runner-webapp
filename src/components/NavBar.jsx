/* /src/components/NavBar.jsx  */

//Libraries
import React, { Component } from 'react';
import {Container} from 'semantic-ui-react';
import {Link } from 'react-router-dom';


class NavBar extends Component {
  render() {
    return (
      <Container fluid>
        <div className="placeholder"/>
        <div className="navContainer">
          <NavLink to="/">
            <img  className="nav-img" src={require('../media/assets/Logo.png')} alt="logo" />
          </NavLink>
          <NavLink exact activeClassName="activeLink" to="/" className="link" >Home</NavLink>
          <NavLink activeClassName="activeLink" to="/Configurations" className="link">Configuration List</NavLink>
          <NavLink activeClassName="activeLink" to="/Evaluations" className="link">Evaluation List</NavLink>
        </div>
      </Container>
    );
  }
}

export default NavBar;
