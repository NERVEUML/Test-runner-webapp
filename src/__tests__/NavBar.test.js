/* /src/__tests__/NavBar.test.js  */

//Libraries
import React from 'react';
import { shallow } from 'enzyme'
//Component To Be Tested
import NavBar from '../components/NavBar.jsx';

//Call to create a Shallow rendering of component
const wrapper = shallow(<NavBar />)

describe('NavBar Component', () => {
  test('render', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot();
  })
});