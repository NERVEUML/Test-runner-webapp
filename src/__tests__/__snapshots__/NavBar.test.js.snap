// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="placeholder"
    />
    <div
        className="navContainer"
    >
        <NavLink
            activeClassName="activeLink"
            className="link"
            exact={true}
            to="/"
        >
            Home
        </NavLink>
        <NavLink
            activeClassName="activeLink"
            className="link"
            to="/Configurations"
        >
            Configuration List
        </NavLink>
        <NavLink
            activeClassName="activeLink"
            className="link"
            to="/Evaluations"
        >
            Evaluation List
        </NavLink>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="placeholder"
      />
      <div
            className="navContainer"
      >
            <NavLink
                  activeClassName="activeLink"
                  className="link"
                  exact={true}
                  to="/"
            >
                  Home
            </NavLink>
            <NavLink
                  activeClassName="activeLink"
                  className="link"
                  to="/Configurations"
            >
                  Configuration List
            </NavLink>
            <NavLink
                  activeClassName="activeLink"
                  className="link"
                  to="/Evaluations"
            >
                  Evaluation List
            </NavLink>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NavBar />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NavBar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="placeholder"
          />
          <div
                    className="navContainer"
          >
                    <NavLink
                              activeClassName="activeLink"
                              className="link"
                              exact={true}
                              to="/"
                    >
                              Home
                    </NavLink>
                    <NavLink
                              activeClassName="activeLink"
                              className="link"
                              to="/Configurations"
                    >
                              Configuration List
                    </NavLink>
                    <NavLink
                              activeClassName="activeLink"
                              className="link"
                              to="/Evaluations"
                    >
                              Evaluation List
                    </NavLink>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="placeholder"
          />
          <div
                    className="navContainer"
          >
                    <NavLink
                              activeClassName="activeLink"
                              className="link"
                              exact={true}
                              to="/"
                    >
                              Home
                    </NavLink>
                    <NavLink
                              activeClassName="activeLink"
                              className="link"
                              to="/Configurations"
                    >
                              Configuration List
                    </NavLink>
                    <NavLink
                              activeClassName="activeLink"
                              className="link"
                              to="/Evaluations"
                    >
                              Evaluation List
                    </NavLink>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NavBar />,
}
`;
